### PR TITLE
Add `keys balance` command to query the balance for a key

### PR DESCRIPTION
**Pull request by ljoss17**
Closes: #912

## Description

This PR adds the sub command 'balance' to the command 'keys' to Hermes.

Usage :
```
hermes keys balance <CHAIN_ID> [--key-name KEY_NAME]
```

______

### PR author checklist:
- [x] Added changelog entry, using [`unclog`](https://github.com/informalsystems/unclog).
- ~[ ] Added tests: integration (for Hermes) or unit/mock tests (for modules).~
  - Following a chat with Soares, this feature doesn't seem to need integration testing, and CLI aren't generally tested except for a few in the E2E python tests.
- [x] Linked to GitHub issue.
- [x] Updated code comments and guide (e.g., `guide/`).

### Reviewer checklist:

- [ ] Reviewed `Files changed` in the GitHub PR explorer.
- [ ] Manually tested (in case integration/unit/mock tests are absent).

### CODE REVIEW

File: relayer-cli/src/commands/keys/balance.rs [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+        let rt = Arc::new(TokioRuntime::new().unwrap());
+        let chain = CosmosSdkChain::bootstrap(chain_config.clone(), rt)
+            .unwrap_or_else(exit_with_unrecoverable_error);
+
+        match chain.query_balance() {
+            Ok(balance) => {
+                Output::success_msg(format!("balance {} {}", balance.amount, balance.denom)).exit()
+            }
+            Err(e) => Output::error(format!("{}", e)).exit(),
```
Comment (seanchen1991) [inline relayer-cli/src/commands/keys/balance.rs:42]: We should perhaps add some additional context to this error message, something like:
```suggestion
            Err(e) => Output::error(format!("there was a problem querying the chain balance: {}", e)).exit(),
```

File: relayer-cli/src/commands/keys/balance.rs [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+use abscissa_core::{Command, Runnable};
+use alloc::sync::Arc;
+use tokio::runtime::Runtime as TokioRuntime;
+
+use ibc::core::ics24_host::identifier::ChainId;
+use ibc_relayer::chain::{ChainEndpoint, CosmosSdkChain};
+
+use crate::application::app_config;
+use crate::conclude::{exit_with_unrecoverable_error, Output};
+
```
Comment (seanchen1991) [inline relayer-cli/src/commands/keys/balance.rs:11]: I think reiterating the documentation for this command in a doc comment for the struct helps other developers, i.e. they won't need to go and browse the guide to figure out what this command does. 

Here's an example from another command:
https://github.com/informalsystems/ibc-rs/blob/01145a1c48c0dc6e3e437492ba80d33962bcb65b/relayer-cli/src/commands/create/channel.rs#L27-L43

**Review by seanchen1991 — COMMENTED** [review]
Left a couple of suggestions regarding messaging and documentation, but otherwise it looks good! 🙂

**Review by seanchen1991 — APPROVED** [review]

File: relayer-cli/src/commands/keys/balance.rs [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+/// 
+/// If successful the balance and denominator of the account, associated with the given keyname
+/// on the given chain, will be displayed.
+#[derive(Clone, Command, Debug, Parser)]
+pub struct KeyBalanceCmd {
+    #[clap(required = true, help = "identifier of the chain")]
+    chain_id: ChainId,
+
+    #[clap(required = true, help = "name of the key")]
+    keyname: String,
```
[the remark this answers is not in the document]
Comment (ljoss17) [reply relayer-cli/src/commands/keys/balance.rs:26]: Indeed, I realized this after I updated the error log of the command.
What I am not clear about is the usage of the query_balance method which takes the wallet name from the configuration file. If it is useful for many cases, I would tend to be more favorable for the option 2.

File: relayer-cli/src/commands/keys/balance.rs [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+                "chain {} not found in configuration file",
+                self.chain_id
+            ))
+            .exit(),
+            Some(chain_config) => chain_config,
+        };
+
+        let rt = Arc::new(TokioRuntime::new().unwrap());
+        let chain = CosmosSdkChain::bootstrap(chain_config.clone(), rt)
+            .unwrap_or_else(exit_with_unrecoverable_error);
```
Comment (romac) [inline relayer-cli/src/commands/keys/balance.rs:44]:
```suggestion
       spawn_chain_runtime(config, &self.chain_id).unwrap_or_else(exit_with_unrecoverable_error);
```

File: relayer-cli/src/commands/keys/balance.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (ljoss17) [reply relayer-cli/src/commands/keys/balance.rs]: The option to modify the query_balance method to accept a key name as an Option. If None is given, the account is retrieved from the configuration file.

File: guide/src/commands/keys/index.md [hunk 5]
```diff
@@ -242,3 +242,24 @@ If the command is successful a message similar to the one below will be displaye
...
+```shell
+USAGE:
+    hermes keys balance <CHAIN_ID> <KEYNAME>
+
+DESCRIPTION:
+    Query balance for a key from a configured chain
+
+POSITIONAL ARGUMENTS:
+    chain_id                  identifier of the chain
+    keyname                   key name
```
Comment (romac) [inline guide/src/commands/keys/index.md:258]:
```suggestion
    key_name                  key name
```

File: guide/src/commands/keys/index.md [hunk 6]
```diff
@@ -242,3 +242,24 @@ If the command is successful a message similar to the one below will be displaye
...
+```
+
+If the command is successful a message with the following format will be displayed:
+
+```
+Success: balance <amount> <denom>
+```
```
Comment (romac) [inline guide/src/commands/keys/index.md:265]: Can you please also add an example when the `--json` flag is supplied?

File: relayer-cli/src/commands/keys/balance.rs [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+///
+/// If successful the balance and denominator of the account, associated with the given keyname
+/// on the given chain, will be displayed.
+#[derive(Clone, Command, Debug, Parser)]
+pub struct KeyBalanceCmd {
+    #[clap(required = true, help = "identifier of the chain")]
+    chain_id: ChainId,
+
+    #[clap(required = true, help = "name of the key")]
+    key_name: String,
```
Comment (romac) [inline relayer-cli/src/commands/keys/balance.rs:25]: This flag should be optional and use the default key when omitted. Please mention it as well in the help message and in the guide.

**Review by romac — CHANGES_REQUESTED** [review]
Very nice! Left a couple suggestions for improvement but otherwise it looks great 🎉

**Review by romac — APPROVED** [review]
Great work!

**Pull request merged**